### PR TITLE
Non-LDAP login issue

### PR DESCRIPTION
**Pull request by shilpaexaltcg**

### CODE REVIEW

**Comment by sylvestre** [comment]
Thanks for fixing the ownership issue
However, you didn't fix when the ldap access is used. I am getting:
```
SyntaxError: JSON.parse: unexpected character at line 1 column 1 of the JSON data
```

https://github.com/mozilla/bugzilla-dashboard/pull/80#issuecomment-489206553

File: src/views/Main/index.jsx [hunk 1]
```diff
@@ -78,8 +79,15 @@ class MainContainer extends Component {
     }
 
     async getReportees(userSession, ldapEmail) {
-      const secretsClient = userSession.getTaskClusterSecretsClient();
-      const partialOrg = await getAllReportees(secretsClient, ldapEmail);
+      let partialOrg;
+      // For LDAP users
+      console.log('login type', userSession.loginType);
+      if (userSession.loginType === 'auth.mozilla.auth0.com') {
```
Comment (armenzg) [inline src/views/Main/index.jsx]: You can check if `process.env.ALTERNATIVE_AUTH` is either true or false to determine if we're using the alternative auth that allows Github authentication for non-LDAP users.

Comment (shilpaexaltcg) [reply src/views/Main/index.jsx]: Thanks @armenzg. I'll try that

File: src/components/auth/auth0.js [hunk 2]
```diff
@@ -9,7 +9,8 @@ export function userSessionFromAuthResult(authResult) {
   return UserSession.fromOIDC({
     oidcProvider: 'mozilla-auth0',
     accessToken: authResult.accessToken,
-    fullName: authResult.idTokenPayload.name,
+    fullName: authResult.idTokenPayload.name === '' ? authResult.idTokenPayload.nickname : authResult.idTokenPayload.name,
+    // loginType: authResult.idTokenPayload.iss,
```
Comment (sylvestre) [inline src/components/auth/auth0.js]:
```suggestion
```

File: src/utils/getAllReportees.js [hunk 3]
```diff
@@ -49,8 +50,17 @@ const findReportees = (completeOrg, email) => {
...
 
-const getAllReportees = async (secretsClient, ldapEmail) => {
-  const completeOrg = await getOrgChart(secretsClient);
+const getAllReportees = async (userSession, ldapEmail) => {
+  let people;
+  if (process.env.ALTERNATIVE_AUTH) {
+    // if non-LDAP user, get fake data
+    people = await (await fetch('people.json')).json();
+  } else {
+    const secretsClient = userSession.getTaskClusterSecretsClient();
```
Comment (sylvestre) [inline src/utils/getAllReportees.js]:
```suggestion
    // LDAP user, retrieve data from the taskcluster secret
    const secretsClient = userSession.getTaskClusterSecretsClient();
```

File: src/views/Main/index.jsx [hunk 4]
```diff
@@ -182,29 +172,34 @@ class MainContainer extends Component {
...
-          this.setState({ teamComponents });
-        }
-      });
+    async teamsData(userSession, partialOrg) {
+      let teamComponents = {};
+      if (userSession.oidcProvider === 'mozilla-auth0') {
+        // if non-LDAP user, get fake data
+        teamComponents = TEAMS_CONFIG;
+      } else {
+        Object.entries(TEAMS_CONFIG).map(async ([teamKey, teamInfo]) => {
```
Comment (sylvestre) [inline src/views/Main/index.jsx]:
```suggestion
        // LDAP user, get the actual data
        Object.entries(TEAMS_CONFIG).map(async ([teamKey, teamInfo]) => {
```

**Pull request merged**